broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/perf/TestState.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.perf;
+
+import io.camunda.zeebe.db.ConsistencyChecksSettings;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDbFactory;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.db.impl.rocksdb.RocksDbConfiguration;
+import io.camunda.zeebe.db.impl.rocksdb.ZeebeRocksDbFactory;
+import io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransactionDb;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.scheduler.ActorScheduler;
+import io.camunda.zeebe.snapshots.ConstructableSnapshotStore;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotStoreFactory;
+import io.camunda.zeebe.util.FileUtil;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+import org.agrona.CloseHelper;
+import org.agrona.concurrent.UnsafeBuffer;
+
+final class TestState {
+
+  TestContext generateContext(final int keyCount, final int keySize, final int valueSize)
+      throws Exception {
+    final var tempDirectory = Files.createTempDirectory("statePerf");
+    final var actorScheduler =
+        ActorScheduler.newActorScheduler()
+            .setIoBoundActorThreadCount(1)
+            .setCpuBoundActorThreadCount(1)
+            .build();
+    actorScheduler.start();
+
+    final var storeFactory = new FileBasedSnapshotStoreFactory(actorScheduler, 1);
+    final var snapshotStore = createSnapshotStore(storeFactory, tempDirectory);
+    takeSnapshot(snapshotStore, keyCount, keySize, valueSize);
+
+    return new TestContext(
+        actorScheduler, tempDirectory, snapshotStore, createDbFactory(), storeFactory);
+  }
+
+  private void takeSnapshot(
+      final ConstructableSnapshotStore snapshotStore,
+      final int keyCount,
+      final int keySize,
+      final int valueSize) {
+    final var snapshot = snapshotStore.newTransientSnapshot(1, 1, 1, 1).get();
+    snapshot.take(path -> takeSnapshot(path, keyCount, keySize, valueSize)).join();
+    snapshot.persist().join();
```

Comment:
❓ I guess we need to use these methods in order to correctly recover, since it creates checksums etc?

Reply:
Correct, and it makes sure the snapshot store sees it as the latest snapshot, and its in memory state is set up properly.